app/models/edition.rb

```diff
@@ -604,19 +601,17 @@ def body_required?
   end
 
   def has_associated_needs?
-    associated_needs.any?
+    respond_to?(:need_ids) && need_ids.try(:any?)
   end
 
   def associated_needs
```

Comment:
I've collected all the needs related code in one place now, and removed `has_associated_needs?`.